server/gitrest/packages/gitrest-base/src/utils/gitWholeSummaryManager.ts

```diff
@@ -68,11 +101,96 @@ function getSummaryObjectFromWholeSummaryTreeEntry(entry: WholeSummaryTreeEntry)
     throw new NetworkError(400, `Unknown entry type: ${entry.type}`);
 }
 
+interface IFullGitTree {
+    tree: ITree;
+    blobs: Record<string, IBlob>;
+}
+function containsFullGitTree(gitTree: ITree): boolean {
+    return gitTree.tree.length === 1
+        && gitTree.tree[0].type === "blob"
+        && gitTree.tree[0].path.endsWith(fullTreePath);
+}
+async function buildFullGitTreeFromGitTree(
+    gitTree: ITree,
+    repoManager: IRepositoryManager,
+): Promise<IFullGitTree> {
+    const blobPs: Promise<IBlob>[] = [];
+    gitTree.tree.forEach((treeEntry) => {
+        if (treeEntry.type === "blob") {
+            blobPs.push(repoManager.getBlob(treeEntry.sha));
+        }
+    });
+    const blobs = await Promise.all(blobPs);
+    const blobMap = {};
+    blobs.forEach((blob) => blobMap[blob.sha] = blob);
+    return {
+        tree: gitTree,
+        blobs: blobMap,
+    };
+}
+async function parseGitTreeContainingFullGitTree(
+    gitTree: ITree,
+    repoManager: IRepositoryManager,
+): Promise<IFullGitTree> {
+    const fullGitTreeBlob = await repoManager.getBlob(
+        gitTree.tree[0].sha
+    );
+    return JSON.parse(Buffer.from(
+        fullGitTreeBlob.content,
+        fullGitTreeBlob.encoding === "base64" ? "base64" : "utf-8",
+    ).toString("utf-8")) as IFullGitTree;
+}
+function convertGitBlobToSummaryBlob(blob: IBlob): IWholeFlatSummaryBlob {
+    return {
+        content: blob.content,
+        encoding: blob.encoding === "base64" ? "base64" : "utf-8",
+        id: blob.sha,
+        size: blob.size,
+    };
+}
+function convertFullGitTreeToFullSummaryTree(
```

Comment:
This file is where the majority of the changes are. I don't like how messy it's getting